google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java

```diff
@@ -287,7 +292,44 @@ public void run() {
           });
     }
 
-    return publishResult;
+    return new ListenableFutureDelegate<String>(publishResult);
+  }
+
+  private static class ListenableFutureDelegate<V> extends SimpleForwardingListenableFuture<V>
```

Comment:
This is copying what's in GAX, right? At least document it. We'll be able to get rid of it later, right?